### PR TITLE
Care reports - various fixes

### PR DESCRIPTION
**Pull request by kkrampa**
I modified model so data need to be reindexed after deploy. I will create a ticket for this.

Ticket for reindex: http://manage.dimagi.com/default.asp?230124

### CODE REVIEW

**Comment by kkrampa** [comment]
@czue

File: custom/care_pathways/sqldata.py [hunk 1]
```diff
@@ -367,30 +380,49 @@ def columns(self):
...
-        for domain in groupped_headers:
-            groupped_practices = [list(v) for l,v in groupby(sorted(domain, key=lambda x:x[3]), lambda x: x[3])]
-            domain_group = DataTablesColumnGroup(self.group_name_fn(domain[0][2]))
-            for practice in groupped_practices:
-                domain_group.add_column(DataTablesColumn(self.group_name_fn(practice[0][3])))
+    @property
+    def domains_with_practices(self):
+        practices = self.config.get('practices')
+        if not practices:
+            practices = ('0', )
```
Comment (czue) [inline custom/care_pathways/sqldata.py]: maybe worth a comment explaining this pattern?

Comment (czue) [inline custom/care_pathways/sqldata.py]: actually i think `'0'` just a magic constant being used to check for `practices` and `domains` being empty? can you not replace that with just assigning them to `None` and checking below using `is None`?

File: custom/care_pathways/utils.py [hunk 2]
```diff
@@ -2,12 +2,13 @@
...
 import json
+
+from corehq.util.quickcache import quickcache
 from dimagi.ext.jsonobject import JsonObject, StringProperty, ListProperty, DictProperty
 from corehq.apps.reports.sqlreport import DataFormatter
-from dimagi.utils.decorators.memoized import memoized
 
 
-@memoized
+@quickcache([], timeout=5 * 60)
```
Comment (czue) [inline custom/care_pathways/utils.py]: I think should have `vary_on='domain'`, no?

**Comment by czue** [comment]
these are tough to follow! i think the quickcache thing is a real problem, other stuff not blocking. didn't review stuff super closely, but assume you are comfortable with the level of QA in terms of having these changes deployed?

**Comment by kkrampa** [comment]
@czue Updated. I tested this locally and it should be good to merge.

**Comment by kkrampa** [comment]
@czue Could you ask someone to run this Ticket for reindex: http://manage.dimagi.com/default.asp?230124 after deploy?

**Pull request merged**